pinot-controller/src/test/java/org/apache/pinot/controller/api/resources/PinotLogicalTableResourceTest.java

```diff
@@ -199,15 +200,46 @@ public void testLogicalTableValidationTests()
     }
   }
 
+  @Test
+  public void testLogicalTableSchemaValidation()
+      throws IOException {
+    String addLogicalTableUrl = _controllerRequestURLBuilder.forLogicalTableCreate();
+    List<String> physicalTableNamesWithType = createHybridTables(List.of("test_table_3"));
+
+    // Test logical table schema does not exist
+    LogicalTableConfig logicalTableConfig =
+        getDummyLogicalTableConfig(LOGICAL_TABLE_NAME, physicalTableNamesWithType, BROKER_TENANT);
+    try {
+      ControllerTest.sendPostRequest(addLogicalTableUrl, logicalTableConfig.toSingleLineJsonString(), getHeaders());
+      fail("Logical Table POST request should have failed");
+    } catch (IOException e) {
+      assertTrue(e.getMessage().contains("Reason: Schema with same name as logical table '" + LOGICAL_TABLE_NAME
+          + "' does not exist"), e.getMessage());
+    }
```

Comment:
nit: could you use `Assert.assertThrows` instead?

Reply:
assertThrows(Exception) would not help as all the error are converted into IOException so we need to differentiate between errors.

For assertThrows(String, Exception) we would need to match the complete string, so I just used it.

Reply:
The IOException conversion happen by the `ControllerTest.sendPostRequest`

Reply:
`assertThrows` also returns the `Exception` so you can make the same assertions on the message string as you're doing here already?

Reply:
I believe, you mean `expectThrows`. Final outcome:

```
    Throwable throwable1 = expectThrows(IOException.class, () -> {
      LogicalTableConfig tableConfig = getDummyLogicalTableConfig(LOGICAL_TABLE_NAME, List.of(), BROKER_TENANT);
      ControllerTest.sendPostRequest(_addLogicalTableUrl, tableConfig.toSingleLineJsonString(), getHeaders());
    });
    assertTrue(throwable1.getMessage().contains("Reason: 'physicalTableConfigMap' should not be null or empty"),
        throwable1.getMessage());
```

Reply:
@yashmayya  In the https://github.com/apache/pinot/pull/15720 I am already refactoring the tests into individual tests. I will make this change as well. We can get this in as is.